source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -182,1257 +182,1274 @@ namespace
         }
     }
 
-    template<class T>
-    void createInterface(const ElementPtr& src, T schema)
+    void writeTargetDef(const PvtPrim* src, DocumentPtr dest, const RtWriteOptions* options)
```

Comment:
Most of this change is simply reordering some methods. readLookInformation calls writeDocument which needed to be defined before use, along with the methods it depends on. The actual work in this code that is being done is in readLookInformation.